File: python/examples/zcsv_create.py
from __future__ import annotations

import sys

from zippedcsv import ZippedCsv


def main() -> int:
    if len(sys.argv) < 2:
        print("Usage: python examples/zcsv_create.py <path_to_zipped_csv_file>")
        return 1

    try:
        zcsv = ZippedCsv(sys.argv[1])
        zcsv.Save()
        zcsv.Close()
        print(f"Zipped CSV file created: {sys.argv[1]}")
        return 0
    except Exception as error:
        print(f"Error: {error}")
        return 1


if __name__ == "__main__":
    raise SystemExit(main())

File: php/examples/zcsv_create.php
<?php
/**
 * Example of using ZippedCsv to create an empty zipped csv file.
 * Usage:
 * php examples/zcsv_create.php <path_to_zipped_csv_file>
 */

require_once __DIR__ . '/../src/mc/csv.php';
require_once __DIR__ . '/../src/mc/zippedcsv.php';

use \mc\ZippedCsv;

if ($argc < 2) {
    echo "Usage: php examples/zcsv_create.php <path_to_zipped_csv_file>" . PHP_EOL;
    exit(1);
}

try {
    $zcsv = new ZippedCsv($argv[1]);
    $zcsv->Save();
    $zcsv->Close();
    echo "Zipped CSV file created: " . $argv[1] . PHP_EOL;
} catch (Exception $e) {
    echo "Error: " . $e->getMessage() . PHP_EOL;
    exit(1);
}
